scvelo/tools/_em_model.py

```diff
@@ -158,15 +168,17 @@ def _prepare_genes(self):
             var_names = var_names[np.argsort(np.sum(X, 0))[::-1][: self._n_top_genes]]
         self._var_names = var_names
 
-    # TODO: Implement
     def state_dict(self):
         """Return the state of the model."""
-        raise NotImplementedError
+        return asdict(self._state_dict)
 
-    # TODO: Implement
-    def export_results_adata(self):
-        """Export the results to the AnnData object."""
-        raise NotImplementedError
+    def export_results_adata(self, copy: bool = True, add_key: str = "fit"):
```

Comment:
See my comment for the `state_dict` function: These functions should be implemented in a separate PR for maximal readability, etc.